### PR TITLE
Overlaid state

### PR DESCRIPTION
**Pull request by ChengshuLi**
- implement overlaid checking function
- add contact report for cloth prims

### CODE REVIEW

File: omnigibson/examples/object_states/overlaid_demo.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    # Create the environment
+    env = og.Environment(configs=cfg, action_timestep=1 / 60., physics_timestep=1 / 60.)
+
+    # Grab object references
+    carpet = env.scene.object_registry("name", "carpet")
+    breakfast_table = env.scene.object_registry("name", "breakfast_table")
+
+    max_steps = 100 if short_exec else -1
+    steps = 0
```
Comment (cremebrule) [inline omnigibson/examples/object_states/overlaid_demo.py:54]: Very cool! Maybe it'd be helpful to print out a user-friendly mesage telling them what they're supposed to do, and what things should cause a change in the Overlaid state?

File: omnigibson/object_states/overlaid.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import itertools
+
+# Create settings for this module
+m = create_module_macros(module_path=__file__)
+
+# Percentage of xy-plane of the object's base aligned bbox that needs to covered by the cloth
+m.OVERLAP_AREA_PERCENTAGE = 0.5
+
+# Subsample cloth particle points to fit a convex hull for efficiency purpose
+m.N_POINTS_CONVEX_HULL = 1000
```
Comment (cremebrule) [inline omnigibson/object_states/overlaid.py:21]: Curious, what's the overall speed with using 1000 points? I'm guessing you chose this value as the empirically-validated optimum trading off fidelity and efficiency?

File: omnigibson/object_states/overlaid.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class Overlaid(KinematicsMixin, RelativeObjectState, BooleanState):
+    @staticmethod
+    def get_dependencies():
+        return KinematicsMixin.get_dependencies() + RelativeObjectState.get_dependencies() + [Touching]
+
+    def _set_value(self, other, new_value, use_ray_casting_method=False):
+        raise NotImplementedError("Overlaid state currently does not support setting.")
+
+    def _get_value(self, other):
```
Comment (cremebrule) [inline omnigibson/object_states/overlaid.py:31]: Might be helpful (actually, in general, for other complex object states) providing docstring here describing high-level logic of what determines whether the obj is overlaid or not. What is the HalfSpaceIntersection checking, exactly? (sorry, I'm not too familiar with the trimesh terms)

File: omnigibson/prims/cloth_prim.py [hunk 4]
```diff
@@ -130,6 +132,36 @@ def particle_positions(self, pos):
         p_local = (r.T @ (pos - t).T).T / s
         self.set_attribute(attr="points", val=array_to_vtarray(arr=p_local, element_type=Gf.Vec3f))
```
Comment (cremebrule) [inline omnigibson/prims/cloth_prim.py:133]: fyi, sorry, this API is super slow. There's a much more optimal API you should call, `Vt.Vec3fArray.FromNumpy(input_array)`

File: omnigibson/prims/cloth_prim.py [hunk 5]
```diff
@@ -130,6 +132,36 @@ def particle_positions(self, pos):
...
 
+    def contact_list(self):
```
Comment (cremebrule) [inline omnigibson/prims/cloth_prim.py:135]: this is cool!

**Review by cremebrule — COMMENTED** [review]
This is super cool!! Left a few comments, but nothing too crazy, just some documentation requests mostly. Otherwise LGTM!

File: omnigibson/examples/object_states/overlaid_demo.py (hunk 1 above)
Comment (ChengshuLi) [reply omnigibson/examples/object_states/overlaid_demo.py:54]: done

File: omnigibson/object_states/overlaid.py (hunk 2 above)
Comment (ChengshuLi) [reply omnigibson/object_states/overlaid.py:21]: overall speed with using 1000 points is ~0.5 ms. Yes I choose this value as the empirically-validated optimum trading off fidelity and efficiency.

File: omnigibson/object_states/overlaid.py (hunk 3 above)
Comment (ChengshuLi) [reply omnigibson/object_states/overlaid.py:31]: Good call! Added more detailed comments and docstring.

File: omnigibson/prims/cloth_prim.py (hunk 4 above)
Comment (ChengshuLi) [reply omnigibson/prims/cloth_prim.py:133]: Did you call this new API in your refactoring PR? Can you modify this in your PR? I will like to keep this PR small for just the overlaid state if possible.

**Comment by ChengshuLi** [comment]
@cremebrule I addressed your comments. PTAL. Thanks!

**Review by cremebrule — APPROVED** [review]
LGTM!

**Pull request merged**